src/Data/Vec/Functional/Algebra/Properties.agda

```diff
@@ -0,0 +1,173 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Some Vector-related module properties
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+open import Function using (_$_)
+open import Data.Product hiding (map)
+open import Data.Nat using (ℕ)
+open import Data.Fin using (Fin)
+open import Data.Vec.Functional
+open import Algebra.Core
+open import Algebra.Bundles
+open import Algebra.Module
+open import Relation.Binary
+import Data.Vec.Functional.Relation.Binary.Equality.Setoid as VecSetoid
+import Algebra.Definitions as AD
+import Algebra.Structures as AS
+import Data.Vec.Functional.Algebra.Base as VFA
+
+module Data.Vec.Functional.Algebra.Properties
+  {c ℓ} (ring : Ring c ℓ) where
+
+private variable
+  n : ℕ
+
+open Ring ring
+open VecSetoid setoid
+open VFA ring
+module SR = Semiring semiring
+open module AD' {n} = AD (_≈ᴹ_ {n})
+open module AS' {n} = AS (_≈ᴹ_ {n})
+open module LD {n} = LeftDefs Carrier (_≈ᴹ_ {n}) using (Congruent)
+
+------------------------------------------------------------------------
+-- Algebraic properties of _+ᴹ_ -ᴹ_ _*ₗ_
+
++ᴹ-cong : Congruent₂ (_+ᴹ_ {n})
++ᴹ-cong x≈y u≈v _ = +-cong (x≈y _) (u≈v _)
+
++ᴹ-assoc : Associative (_+ᴹ_ {n})
++ᴹ-assoc _ _ _ _ = +-assoc _ _ _
```

Comment:
I think it would be clearer to explicitly write out the arguments here and in other definitions, and not use wildcards and inference.

Reply:
solved